stumpy/stumped.py

```diff
@@ -205,6 +204,206 @@ def _dask_stumped(
+        will contain exactly 2 * k + 2 columns. The first k columns (i.e., out[:, :k])
+        consists of the top-k matrix profile, the next set of k columns
+        (i.e., out[:, k:2k]) consists of the corresponding top-k matrix profile
+        indices, and the last two columns (i.e., out[:, 2k] and out[:, 2k+1] or,
+        equivalently, out[:, -2] and out[:, -1]) correspond to the top-1 left
+        matrix profile indices and the top-1 right matrix profile indices, respectively.
+    """
+    core.check_ray(ray_client)
+
+    n_A = T_A.shape[0]
+    n_B = T_B.shape[0]
+    l = n_A - m + 1
+
+    nworkers = core.get_ray_nworkers(ray_client)
+
+    ndist_counts = core._count_diagonal_ndist(diags, m, n_A, n_B)
+    diags_ranges = core._get_array_ranges(ndist_counts, nworkers, False)
+    diags_ranges += diags[0]
+
+    # Put data in the Ray object store
+    T_A_ref = ray_client.put(T_A)
+    T_B_ref = ray_client.put(T_B)
+    M_T_ref = ray_client.put(M_T)
+    μ_Q_ref = ray_client.put(μ_Q)
+    Σ_T_inverse_ref = ray_client.put(Σ_T_inverse)
+    σ_Q_inverse_ref = ray_client.put(σ_Q_inverse)
+    M_T_m_1_ref = ray_client.put(M_T_m_1)
+    μ_Q_m_1_ref = ray_client.put(μ_Q_m_1)
+    T_A_subseq_isfinite_ref = ray_client.put(T_A_subseq_isfinite)
+    T_B_subseq_isfinite_ref = ray_client.put(T_B_subseq_isfinite)
+    T_A_subseq_isconstant_ref = ray_client.put(T_A_subseq_isconstant)
+    T_B_subseq_isconstant_ref = ray_client.put(T_B_subseq_isconstant)
+
+    diags_refs = []
+    for i in range(nworkers):
+        diags_ref = ray_client.put(
+            np.arange(diags_ranges[i, 0], diags_ranges[i, 1], dtype=np.int64),
+        )
+        diags_refs.append(diags_ref)
+
+    ray_stump_func = ray_client.remote(core.deco_ray_tor(_stump))
+
+    refs = []
+    for i in range(nworkers):
+        refs.append(
+            ray_stump_func.remote(
+                T_A_ref,
+                T_B_ref,
+                m,
+                μ_Q_ref,
+                M_T_ref,
+                σ_Q_inverse_ref,
+                Σ_T_inverse_ref,
+                μ_Q_m_1_ref,
+                M_T_m_1_ref,
+                T_A_subseq_isfinite_ref,
+                T_B_subseq_isfinite_ref,
+                T_A_subseq_isconstant_ref,
+                T_B_subseq_isconstant_ref,
+                diags_refs[i],
+                ignore_trivial,
+                k,
+            )
+        )
+
+    results = ray_client.get(refs)
+    profile, profile_L, profile_R, indices, indices_L, indices_R = results[0]
+    # Must make a mutable copy from Ray's object store (ndarrays are immutable)
+    profile = profile.copy()
+    profile_L = profile_L.copy()
+    profile_R = profile_R.copy()
+    indices = indices.copy()
+    indices_L = indices_L.copy()
+    indices_R = indices_R.copy()
+
+    for i in range(1, nworkers):
+        P, PL, PR, I, IL, IR = results[i]
+        # Must make a mutable copy from Ray object store (ndarrays are immutable)
+        P = P.copy()
```

Comment:
@seanlaw 
I am not sure about the computing cost of `.copy()` here but I think we can avoid it in this and the following lines. 
The in-place update of arrays occur only for `profile`, `profile_L`, `profile_R`,  and their corresponding indices.

Reply:
Good point! I will try to remove these copies and see what happens.

Reply:
@NimaSarajpoor I've commented them out and tests are passing now

Reply:
No idea why I missed your messages in this PR. Going to resolve it now!